treeherder/client/thclient/client.py

```diff
@@ -684,6 +686,14 @@ def _get_uri(self, project, endpoint, data=None, oauth_key=None,
 
         return uri
 
+    def _get_revision_id(self, repo_name, revision):
+        """
+        Return the id of a given revision from treeherder
+        """
+        url = "%s/%s/resultset/?revision=%s" % (self.API_ROOT, repo_name, revision)
```

Comment:
As I suggested before, use _get_uri here. Actually, better yet, create a _get_json method here (similiar to the existing _post_json) and use it for getting stuff. We can consolidate various error handling functionality in there.